android/tangram/jni/platform_android.cpp

```diff
@@ -266,5 +276,16 @@ void setCurrentThreadPriority(int priority) {
     setpriority(PRIO_PROCESS, tid, priority);
 }
 
+void featureSelectionCallback(JNIEnv* jniEnv, const std::vector<std::shared_ptr<Tangram::Properties>>& items) {
+
+    jlong jresult = 0;
+    jclass propertiesClass = jniEnv->FindClass("com/mapzen/tangram/Properties");
+
+    jresult = reinterpret_cast<jlong>(new std::shared_ptr<Tangram::Properties >(items[0]));
```

Comment:
We can get @ecgreb's input here. 

I would assume, the API to return a vector of all selected features and the client app to filter on what is picked.
How you are doing for linux and osx: https://github.com/tangrams/tangram-es/blob/feature-nit-picking/osx/src/main.cpp#L53-L57

Its highly likely, if client expects a search result to be picked, but tangram return a poi right adjacent to the search result.